src/character.cpp

```diff
@@ -3122,13 +3122,19 @@ void find_ammo_helper( T &src, const item &obj, bool empty, Output out, bool nes
         } else {
             // Look for containers with any liquid and loose frozen liquids
             src.visit_items( [&src, &nested, &out]( item * node, item * parent ) {
-                if( parent->is_watertight_container() && node->is_frozen_liquid() ) {
+                if( parent != nullptr && parent->is_watertight_container() && node->is_frozen_liquid() ) {
                     return VisitResponse::SKIP;
                 }
 
-                if( ( parent->is_container() && node->made_of( phase_id::LIQUID ) ) || node->is_frozen_liquid() ) {
+                if( node->is_container() && !node->is_container_empty() ) {
+                    return VisitResponse::NEXT;
+                }
```

Comment:
I think this can be achieved by simply passing `nested` as true?

Reply:
You might have a point there. `VisitResponse::NEXT` should probably only be returned when `nested` is true.

I'll take a look at that tomorrow.